File: Wacomi.API/Models/Circles/CircleEventParticipation.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Wacomi.API.Models
{
    public enum CircleEventParticipationStatus {
        Waiting,
        Confirmed,
        Canceled
    }
    public class CircleEventParticipation : IAppUserLinkable
    {
        public CircleEvent CircleEvent { get; set;}
        [Required]
        public int CircleEventId { get; set;} 
        public CircleEventParticipationStatus Status { get; set;} = CircleEventParticipationStatus.Waiting;
        [MaxLength(StaticData.MidTextLength)]
        public string Message{get; set;}
        public AppUser AppUser {get; set;}
        [Required]
        public int? AppUserId {get; set;}
        public DateTime DateCreated {get; set;} = DateTime.Now;
    }
}

File: Wacomi.SPA/src/app/_models/CircleEventParticipation.ts
import { IAppUserLinkable } from "./IAppUserLinkable";

// public CircleEventForReturnDto CircleEvent { get; set;}
//         public int CircleEventId { get; set;} 
//         public CircleEventParticipationStatus Status { get; set;}
//         public string Message{get; set;}
//         public AppUserForReturnDto AppUser {get; set;}
//         public int? AppUserId {get; set;}
//         public DateTime DateCreated {get; set;}

export enum CircleEventParticipationStatus {
    Waiting = 1,
    Confirmed,
    Canceled
}

export interface CircleEventParticipation extends IAppUserLinkable {
    circleEventId: number;
    status: CircleEventParticipationStatus;
    message: string;
    dateCreated?: Date;
}
